app/assets/javascripts/discourse/app/initializers/narrow-desktop.js

```diff
@@ -0,0 +1,37 @@
+import NarrowDesktop from "discourse/lib/narrow-desktop";
+
+export default {
+  name: "narrow-desktop",
+
+  initialize(container) {
+    NarrowDesktop.init();
+    const site = container.lookup("service:site");
+    site.set("narrowDesktopView", NarrowDesktop.narrowDesktopView);
+
+    if ("ResizeObserver" in window) {
+      this._resizeObserver = new ResizeObserver((entries) => {
+        for (let entry of entries) {
+          const oldNarrowDesktopView = site.narrowDesktopView;
+          const newNarrowDesktopView = NarrowDesktop.isNarrowDesktopView(
+            entry.contentRect.width
+          );
+          if (oldNarrowDesktopView !== newNarrowDesktopView) {
+            const applicationController = container.lookup(
+              "controller:application"
+            );
+            site.set("narrowDesktopView", newNarrowDesktopView);
+            applicationController.set("showSidebar", !newNarrowDesktopView);
+            applicationController.appEvents.trigger(
+              "sidebar-toggle:force-refresh"
+            );
+          }
+        }
+      });
+
+      const mainElement = document.querySelector("#main");
```

Comment:
In tests, wouldn't the width of `#main` be very narrow since the ember app is rendered in a smaller element?